### PR TITLE
Add new SearchBar component

### PR DESCRIPTION
**Pull request by Copons**
Moved the `<div class="search-bar">` element from `app.jsx` into a new (Redux connected) `SearchBar` component and Reduxified the `SearchField` component.

This is part of my ES6-Redux conversion effort.

### CODE REVIEW

File: lib/search-bar.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		>
+			<NewNoteIcon />
+		</button>
+	</div>;
+
+const {
+	newNote,
+	toggleNavigation,
+} = appState.actionCreators;
+const { recordEvent } = tracks;
```
Comment (dmsnell) [inline lib/search-bar.jsx]: can we move this up above the React component? it's just a personal style request as I like to keep constant declarations at the top.

File: lib/search-field.jsx [hunk 2]
```diff
@@ -51,10 +53,24 @@ export class SearchField extends Component {
...
 
-const mapStateToProps = ( { appState } ) => ( {
-	query: appState.filter,
-	placeholder: appState.listTitle,
-	searchFocus: appState.searchFocus,
+const {
+	search,
+	setSearchFocus,
+} = appState.actionCreators;
+const { recordEvent } = tracks;
```
Comment (dmsnell) [inline lib/search-field.jsx]: _ibid_. are you comfortable moving this up to the top?

File: lib/search-bar.jsx (hunk 1 above)
Comment (Copons) [inline lib/search-bar.jsx]: Curiously enough, I got to think about this a couple of hours ago!
My reasoning would be to keep the Redux stuff close to each other, but it feels a bit weird indeed.
I'm gonna move them up no problems.

**Comment by Copons** [comment]
@dmsnell reviews don't show as outdated, but they are.

**Comment by dmsnell** [comment]
@Copons if you rebase this agains the recent **master** I'll review it and we can try to get it out there

**Comment by Copons** [comment]
@dmsnell Rebased!

File: lib/search-bar.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		<SearchField />
+		<button
+			className="button button-borderless"
+			disabled={ showTrash }
+			onClick={ onNewNote }
+			title="New Note"
+		>
+			<NewNoteIcon />
+		</button>
+	</div>;
```
Comment (dmsnell) [inline lib/search-bar.jsx]: just a personal preference, and maybe I've broken my own preference, but what do you think about surrounding this in parens just to visually delineate the JSX?

```js
} ) => (
	<div…
	</div>
);
```

File: lib/search-bar.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+const mapStateToProps = ( { appState: state } ) => ( {
+	showTrash: state.showTrash,
+} );
+
+const mapDispatchToProps = ( dispatch, { noteBucket } ) => ( {
+	onNewNote: () => {
+		dispatch( newNote( { noteBucket } ) );
+		recordEvent( 'list_note_created' );
```
Comment (dmsnell) [inline lib/search-bar.jsx]: earmarking since we do this in two places: from the app menu and from here…

**Comment by dmsnell** [comment]
@copons thanks for rebasing. the code looks good here and my tests seemed like they worked

 - was able to search notes and interaction with the search field was normal
 - created a new note from the button
 - created a new note from the app menu

:shipit:

**Pull request merged**